src/component/ContentMessage/index.tsx

```diff
@@ -0,0 +1,144 @@
+import Avatar from "common/Avatar";
+import React, { memo, useEffect, useRef, useState } from "react";
+import { IoCall } from "react-icons/io5";
+import { BsFillCameraVideoFill, BsFileEarmarkImageFill } from "react-icons/bs";
+import {
+  AiOutlineExclamation,
+  AiFillPlusCircle,
+  AiOutlineFileGif,
+  AiOutlineClose,
+} from "react-icons/ai";
+import _ from "lodash";
+
+import * as S from "./styles";
+import { Text } from "common/Text";
+import theme from "constant/theme";
+
+interface ContentMessageProps {}
+
+const ContentMessage: React.FC<ContentMessageProps> = () => {
+  const [message, setMessage] = useState();
+
+  const uploadFileRef = useRef(null);
+  const [fileImages, setImages] = useState<(string | ArrayBuffer | null)[]>([]);
+
+  useEffect(() => {
+    const dropArea = uploadFileRef.current;
+    if (dropArea) {
+      ["dragenter", "dragover", "dragleave", "drop"].forEach((eventName) => {
+        // @ts-ignore
+        dropArea?.addEventListener(eventName, preventDefaults, false);
+      });
+
+      // @ts-ignore
+      dropArea?.addEventListener("drop", handleDrop, false);
+    }
+
+    return () => {
+      if (dropArea) {
+        ["dragenter", "dragover", "dragleave", "drop"].forEach((eventName) => {
+          // @ts-ignore
+          dropArea?.removeEventListener(eventName, preventDefaults, false);
+        });
+        // @ts-ignore
+        dropArea?.removeEventListener("drop", handleDrop, false);
+      }
+    };
+  }, [fileImages]);
+
+  const handleDrop = (e: React.DragEvent) => {
+    const files = e?.dataTransfer.files;
+    handleFiles(files[0]);
+  };
+
+  const handleFiles = (files: any) => {
+    let reader = new FileReader();
+    reader.readAsDataURL(files);
+    reader.onloadend = function () {
+      fileImages.push(reader.result);
+      setImages([...fileImages]);
+    };
+  };
+
+  const preventDefaults = (e: Event) => {
+    e.preventDefault();
+    e.stopPropagation();
+  };
+
+  const handleDeleteImage = (file: any) => {
+    const indexFile = fileImages.findIndex((item) => item === file);
+
+    if (indexFile !== -1) {
+      fileImages.splice(indexFile, 1);
+      setImages([...fileImages]);
```

Comment:
Chỗ này cũng thế, splice nó tác động trực tiếp vào biến gốc, tức ở đây là state fileImages.
Với chỗ này bạn đã dùng findIndex item === file thì t nghĩ xài filter luôn cho gọn

```suggestion
      setImages(images => images.filter(image => image !== file));
```